File: backoffice/src/shared/domain/valueObjects/IntValueObject.py
"""
 *
 * Class
 *
"""

class IntValueObject:

    """
     *
     * Attributes 
     *
    """

    __value : int

    """
     *
     * Methods 
     *
    """

    def __init__( self, value : int ) -> object:
        self.__value = value
    
    def getValue( self ) -> int:
        return self.__value
    
    def equals( self, other : object ) -> bool:
        return self.__value == other.getValue()
    
    def isLessThan( self, other : object ) -> bool:
        return self.__value < other.getValue()
    
    def isGreaterThan( self, other : object ) -> bool:
        return self.__value > other.getValue()
    
    def isLessThanOrEqual( self, other : object ) -> bool:
        return self.__value <= other.getValue()
    
    def isGreaterThanOrEqual( self, other : object ) -> bool:
        return self.__value >= other.getValue()
    
    def toString( self ) -> str:
        return str( self.__value )

File: preparation/src/shared/domain/valueObjects/IntValueObject.cs
/* 
 * 
 * Class
 *
*/

namespace preparation.src.shared.domain {

    public class IntValueObject {

        /* 
         * 
         * Attributes
         *
        */

        private int value;

        /* 
         * 
         * Methods
         *
        */

        public IntValueObject( int value ) {
            this.value = value;
        }

        public int getValue() {
            return value;
        }

        public bool equals( IntValueObject other ) {
            return value == other.getValue();
        }

        public bool isLessThan( IntValueObject other ) {
            return value < other.getValue();
        }

        public bool isGreaterThan( IntValueObject other ) {
            return value > other.getValue();
        }

        public bool isLessThanOrEqual( IntValueObject other ) {
            return value <= other.getValue();
        }

        public bool isGreaterThanOrEqual( IntValueObject other ) {
            return value >= other.getValue();
        }

        public string toString() {
            return value.ToString();
        }

    }

}